lib/base/type.cpp

```diff
@@ -32,6 +38,60 @@ INITIALIZE_ONCE_WITH_PRIORITY([]() {
 	Type::Register(type);
 }, InitializePriority::RegisterTypeType);
 
+static std::vector<Type::Ptr> l_SortedByLoadDependencies;
+static Atomic l_SortingByLoadDependenciesDone (false);
+
+typedef std::unordered_map<Type*, bool> Visited; // https://stackoverflow.com/a/8942986
+
+INITIALIZE_ONCE_WITH_PRIORITY([] {
+	auto types (Type::GetAllTypes());
+
+	types.erase(std::remove_if(types.begin(), types.end(), [](auto& type) {
+		return !ConfigObject::TypeInstance->IsAssignableFrom(type);
+	}), types.end());
```

Comment:
I would also inline this check into the `visit` callback, since you don't have to remove those unwanted types beforehand.

Reply:
But I need to `VERIFY(sorted.size() == types.size());`